classes/local/store/object_file_system.php

```diff
@@ -756,6 +762,30 @@ public function generate_presigned_url_to_external_file($contenthash, $headers =
         return $this->externalclient->generate_presigned_url($contenthash, $headers);
     }
 
+    /**
+     * Check if file extension is whitelisted.
+     * @param string $filename
+     * @return bool
+     * @throws \dml_exception
+     */
+    public function is_extension_whitelisted($filename) {
```

Comment:
this seems to be reinventing the wheel, why can't you just call:

```
$util = new filetypes_util();
$util->is_whitelisted('txt', 'doc txt rtf')
```

Reply:
updated